postreise/plot/plot_bar_generation_stack.py

```diff
@@ -0,0 +1,234 @@
+    :param str/list resources: one or a list of resources. *'curtailment'*,
+        *'solar_curtailment'*, *'wind_curtailment'*, *'wind_offshore_curtailment'*
+        are valid entries together with all available generator types in the area(s).
+        The order of the resources determines the stack order in the figure.
+    :param list/str area_types: list of area_type(s), each area_type is one of
+        *'loadzone'*, *'state'*, *'state_abbr'*, *'interconnect'*, defaults to None.
+    :param list/str scenario_names: list of scenario name(s) of same len as scenario
+        ids, defaults to None.
+    :param bool curtailment_split: if curtailments are split into different
+        categories, defaults to True.
+    :param dict t2c: user specified color of resource type to overwrite type2color
+        default dict. key: resource type, value: color code.
+    :param dict t2l: user specified label of resource type to overwrite type2label
+        default dict. key: resource type, value: label.
+    :param dict t2hc: user specified color of curtailable resource hatches to overwrite
+        type2hatchcolor default dict. key: resource type, valid keys are
+        *'curtailment'*, *'solar_curtailment'*, *'wind_curtailment'*,
+        *'wind_offshore_curtailment'*, value: color code.
+    :param dict titles: user specified figure titles, key: area, value: new figure
+        title in string, use area as title if None.
+    :param bool plot_show: save the generated figure or not, defaults to True.
+    :param bool save: save the generated figure or not, defaults to False.
+    :param dict filenames: user specified filenames, key: area, value: new filename
+        in string, use area as filename if None.
+    :param str filepath: if save is True, user specified filepath, use current
+        directory if None.
+    :return: (*list*) -- matplotlib.axes.Axes object of each plot in a list.
+    :raises TypeError:
+        if resources is not a list and/or
+        if titles is provided but not in a dictionary format and/or
+        if filenames is provided but not in a dictionary format.
+    :raises ValueError:
+        if length of area_types and areas is different and/or
+        if length of scenario_names and scenario_ids is different.
+    """
+    if isinstance(areas, str):
+        areas = [areas]
+    if not scenario_ids:
+        scenario_ids = []
+    if isinstance(scenario_ids, (int, str)):
+        scenario_ids = [scenario_ids]
+    if isinstance(resources, str):
+        resources = [resources]
+    if not isinstance(resources, list):
+        raise TypeError("ERROR: resources should be a list")
+    if isinstance(area_types, str):
+        area_types = [area_types]
+    if not area_types:
+        area_types = [None] * len(areas)
+    if len(areas) != len(area_types):
+        raise ValueError(
+            "ERROR: if area_types are provided, it should have the same number of entries with areas"
```

Comment:
`it should have the same number of entries with areas` --> `number of area_types must match number of areas`